monitoring/rid_qualifier/utils.py

```diff
@@ -0,0 +1,129 @@
+from datetime import datetime, timedelta
+
+
+class RIDSerializer():
+    ''' A class to serialize RID objects in the qualifier '''
+        
+    def make_json_compatible(self, struct: Any) -> Any:
+        if isinstance(struct, tuple) and hasattr(struct, '_asdict'):
+            return {k: self.make_json_compatible(v) for k, v in struct._asdict().items()}
+        elif isinstance(struct, dict):
+            return {k: self.make_json_compatible(v) for k, v in struct.items()}
+        elif isinstance(struct, str):
+            return struct
+        try:
+            return [self.make_json_compatible(v) for v in struct]
+        except TypeError:
+            return struct
+
+
+
+class QueryBoundingBox(NamedTuple):
+    ''' This is the object that stores details of query bounding box '''
+
+    name: str
+    shape: Polygon
+    timestamp_before: timedelta
+    timestamp_after: timedelta
+    
+class FlightPoint(NamedTuple):
+    ''' This object holds basic information about a point on the flight track, it has latitude, longitude and altitude in WGS 1984 datum '''
+
+    lat: float # Degrees of latitude north of the equator, with reference to the WGS84 ellipsoid. For more information see: https://github.com/uastech/standards/blob/master/remoteid/canonical.yaml#L1160
+    lng: float # Degrees of longitude east of the Prime Meridian, with reference to the WGS84 ellipsoid. For more information see: https://github.com/uastech/standards/blob/master/remoteid/canonical.yaml#L1170
+    alt: float # meters in WGS 84, normally calculated as height of ground level in WGS84 and altitude above ground level
+    
+
+class AircraftPosition(NamedTuple):
+    ''' A object to hold AircraftPosition details for Remote ID purposes, it mataches the RIDAircraftPosition  per the RID standard, for more information see https://github.com/uastech/standards/blob/36e7ea23a010ff91053f82ac4f6a9bfc698503f9/remoteid/canonical.yaml#L1091'''
+
+    lat : float 
+    lng : float 
+    alt : float
+    accuracy_h : str
+    accuracy_v : str
+    extrapolated : bool
+    pressure_altitude : float
+
+class AircraftHeight(NamedTuple):
+    ''' A object to hold relative altitude for the purposes of Remote ID. For more information see: https://github.com/uastech/standards/blob/36e7ea23a010ff91053f82ac4f6a9bfc698503f9/remoteid/canonical.yaml#L1142 '''
+
+    distance: float
+    reference: str
+
+class AircraftState(NamedTuple):
+    ''' A object to hold Aircraft state details for remote ID purposes. For more information see the published standard API specification at https://github.com/uastech/standards/blob/36e7ea23a010ff91053f82ac4f6a9bfc698503f9/remoteid/canonical.yaml#L1604 '''
+    
+    timestamp: datetime 
+    operational_status: str 
+    position: AircraftPosition # See the definition above 
+    height: AircraftHeight # See the definition above 
+    track: float 
+    speed: float 
+    speed_accuracy: str 
+    vertical_speed: float 
+
+
+class RIDFlight(NamedTuple):
```

Comment:
OK done now, thanks!

Reply:
When I move these to the monitoring.monitorlib.rid, and try to import via `from monitoring.monitorlib import rid` , I am getting a `ModuleNotFoundError: No module named 'monitoring'` error. Does this have be run via Docker, I am not sure why python is giving import errors. I will put it back in the file, we can debug later.

Reply:
It's hard to make sure every user's environment is set up the same way, but Docker does an excellent job of eliminating this difficulty.  I recommend adding a Dockerfile that will build an environment suitable for running these tools regardless of host system; [here](https://github.com/interuss/dss/blob/master/monitoring/tracer/Dockerfile) is a good place to start.  Note that the test_definitions folder will need to be specified as a [bind mount](https://docs.docker.com/storage/bind-mounts/).  Good practice would be to include a short script illustrating how to run the application in Docker; [here](https://github.com/interuss/dss/blob/master/monitoring/tracer/run_poll_locally.sh) is an example of that (this one will be much simpler because there are no command line arguments currently).  Note that the `-v` option is what establishes a bind mount.

For local debugging outside of Docker, the `monitoring` top-level folder must be added to your [PYTHONPATH](https://www.tutorialspoint.com/What-is-PYTHONPATH-environment-variable-in-Python).

Reply:
I have built the docker file and made a first pass of the `run_locally` file, I am confident with the Dockerfile but need to test the .sh file in a bit more detail.

Reply:
OK now both the `run_locally` file and the entry point has been resolved.